### DISCUSSION TITLE
question: chart in winforms .net 6.0

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/winforms#6163] [Q&A]
Hi

In .NET 4.8 I used the chart UI widget in winforms UI designer to make different kind of charts (pie chart, bar chart, ...).
Now I try to port my .net 4.8 code to .net 6.0 but I can't find the chart widget anymore. It's just not in the list...

I use visual studio community 17.0

Thanks

**Comment by RussKie** [comment]
I think you may be referring to the [chart control](https://github.com/dotnet/winforms-datavisualization), which has been deprecated in .NET Core. The source code has been made open source and you're welcome to fork it and make use of it in your project.

**Comment by kirsan31** [comment]
Yea @RussKie pointed you in the right direction. BUT you still can't use chart designer :( Moreover, you will no longer be able to open a form  (with a chart on it) in the designer.  [Details](https://github.com/dotnet/winforms/discussions/5220#discussioncomment-987739).

**Comment by ghost** [comment]
Why are chart controls not implemented in .NET 6.0?

Will there be a replacement chart control added to the UI designer in .NET 6.0 (or any update)?

**Reply by RussKie** [reply]
It was deprecated, and no replacement is planned.

**Reply by saint4eva** [reply]
Try this

https://github.com/microcharts-dotnet/Microcharts

**Reply by SommerEngineering** [reply]
I recommend this: https://github.com/ScottPlot/ScottPlot 🙂

**Comment by kilitary** [comment]
how "Charts" could be deprecated?

p.s.
All charts i found in nuget for NET 6.0 is js based. Drawing **js** charts in **c#** is a little strange. C# is language that takes _all_ aspects of windows desktop apps which is not related to asp/web/js engines.

**Reply by saint4eva** [reply]
You can use this library:

https://github.com/microcharts-dotnet/Microcharts

**Reply by beefydog** [reply]
not for winforms - nice though

**Reply by SommerEngineering** [reply]
I recommend this: https://github.com/ScottPlot/ScottPlot 🙂 Works with WinForms as well.

**Comment by saint4eva** [comment]
> In .NET 4.8 I used the chart UI widget in winforms UI designer to make different kind of charts (pie chart, bar chart, ...). Now I try to port my .net 4.8 code to .net 6.0 but I can't find the chart widget anymore. It's just not in the list...

https://github.com/microcharts-dotnet/Microcharts

**Comment by mcDandy** [comment]
They have emoved multiple controls which were very usable. This one is one of them...

**Comment by kirsan31** [comment]
Guys, your can try my fork: https://github.com/kirsan31/winforms-datavisualization/tree/dev
- .Net6.
- ~~Partially (all major elements work)~~ **Full** new designer support. 
- Lots of performance improvements.

**Reply by paul1956** [reply]
Is there a NuGet? Thanks for doing this. I will try it tomorrow.

**Reply by kirsan31** [reply]
> Is there a NuGet? Thanks for doing this. I will try it tomorrow.

Unfortunately NuGet is missing.

**Reply by RussKie** [reply]
Sometime ago I tinkered with this repo, and I made it build from cli,  updated it to .NET 6, and made it produce the NuGet package as part of the build run. I'll put aside some time to add it to the @kirsan31's fork.

**Reply by RussKie** [reply]
Actually, I have those in my [fork](https://github.com/RussKie/winforms-datavisualization/commits/enhance) - to save time you can pull those yourself. You probably only need the top three commits.
![image](https://user-images.githubusercontent.com/4403806/189844828-1b7cf139-62f6-45a4-aebd-d50fa3806228.png)

**Reply by kirsan31** [reply]
> Actually, I have those in my [fork](https://github.com/RussKie/winforms-datavisualization/commits/enhance) - to save time you can pull those yourself. 

Thanks, I'll be sure to check it out when I have some free time (I have never dealt with publishing NuGet packages).

**Reply by kirsan31** [reply]
@RussKie can you tell pls how you dealed with https://github.com/NuGet/Home/issues/11905?
I see nothing related to it in https://github.com/RussKie/winforms-datavisualization/tree/enhance :(
After changing `TargetFrameworks` to `net6.0-windows;net7.0-windows` I have:

> Error	NU1012	Some included files are included under TFMs which are missing a platform version: net6.0-windows, net7.0-windows

Thanks.

**Reply by RussKie** [reply]
To be honest I can't remember. I'd expect the latest NuGet to convey a better error message (as per https://github.com/NuGet/NuGet.Client/pull/4843).
I can vaguely recall that I have restructured the packaging completely - https://github.com/RussKie/winforms-datavisualization/commit/efd1b5e21ed33fbd361e8805e86b3964503cf9c7 (some bits from https://github.com/RussKie/winforms-datavisualization/commit/52a01aa2f4decb97f4a05a017be81f84eaf79935 may also be relevant).

**Reply by kirsan31** [reply]
@RussKie 
> To be honest I can't remember. I'd expect the latest NuGet to convey a better error message (as per [NuGet/NuGet.Client#4843](https://github.com/NuGet/NuGet.Client/pull/4843)). I can vaguely recall that I have restructured the packaging completely - [RussKie/winforms-datavisualization@efd1b5e](https://github.com/RussKie/winforms-datavisualization/commit/efd1b5e21ed33fbd361e8805e86b3964503cf9c7) (some bits from [RussKie/winforms-datavisualization@52a01aa](https://github.com/RussKie/winforms-datavisualization/commit/52a01aa2f4decb97f4a05a017be81f84eaf79935) may also be relevant).

Yes now they show a better error message.  But that doesn't make it any easier :⁠-⁠(
Ah, I see now:
_Build NuGet without pkg_ commit and `<GeneratePackageOnBuild>True</GeneratePackageOnBuild>`...

---------------
Nvm. Got directions here: https://github.com/gitextensions/GitExtensions.GerritPlugin/pull/46

**Comment by kilitary** [comment]
это всё тесты

**Reply by VirabhadraJ** [reply]
Что именно?
